service/src/main/java/bio/terra/landingzone/common/utils/MetricUtils.java

```diff
@@ -0,0 +1,63 @@
+package bio.terra.landingzone.common.utils;
+
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.Timer;
+import java.time.Duration;
+
+public class MetricUtils {
+  private static final String NAMESPACE = "landingzone";
+  private static final String CLOUD_PLATFORM_TAG = "cloudPlatform";
+  private static final String AZURE_PLATFORM_NAME = "AZURE";
+  private static final String LANDINGZONE_TYPE_TAG = "type";
+  private static final String LANDINGZONE_STEP_TAG = "step";
+
+  // this is to set reasonable upper/lower bound for a step. postgres, aks take 7-8 minutes;
+  // histogram metric would have buckets for all results.
+  private static final int LANDINGZONE_STEP_MAX_DURATION_MILLISECONDS = 60 * 10 * 1000; // 10min
+  private static final int LANDINGZONE_STEP_MIN_DURATION_MILLISECONDS = 60 * 1000; // 1min
+
+  private MetricUtils() {}
+
+  public static void incrementLandingZoneCreation(String type) {
+    Metrics.globalRegistry
+        .counter(
+            String.format("%s.creation.count", NAMESPACE),
+            CLOUD_PLATFORM_TAG,
+            AZURE_PLATFORM_NAME,
+            LANDINGZONE_TYPE_TAG,
+            type)
+        .increment();
+  }
+
+  public static void incrementLandingZoneCreationFailure(String type) {
+    Metrics.globalRegistry
+        .counter(
+            String.format("%s.creation.failure.count", NAMESPACE),
```

Comment:
Should there be a similar .success.count metric?

Reply:
There is 'creation.count' which counts all request. And specific counter for failures.